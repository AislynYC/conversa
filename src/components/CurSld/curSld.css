/* current slide */
#current-sld-container {
  width: 100%;
  margin: 0 auto;
}
#current-sld-border {
  width: 100%;
  margin: 0 auto;
  border-radius: 4px;
  /* prettier-ignore */
  box-shadow: 0px 2px 1px -1px rgba(0, 0, 0, 0.2),0px 1px 1px 0px rgba(0, 0, 0, 0.14),0px 1px 3px 0px rgba(0, 0, 0, 0.12);
  padding-top: 56.25%;
  position: relative;
}

#current-sld {
  box-sizing: border-box;
  padding: 40px;
  font-size: 1.6em;
  background-color: white;
  color: black;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  border-radius: 5px;
}

.member-info {
  position: absolute;
  right: 2%;
  bottom: 3%;
  display: flex;
}

.hand-group {
  display: flex;
  margin-right: 10px;
}

#hand-icon {
  margin-right: 5px;
}

#member-icon {
  margin-right: 5px;
}

.qus-div {
  word-break: break-all;
  line-height: 2.5rem;
  position: absolute;
  z-index: 99;
}

#current-sld:fullscreen > .qus-div {
  line-height: 4rem;
}

.opt-ul > li {
  padding: 0.5em 0;
}

.reaction-count {
  margin-left: 10px;
}

.reaction-icon.scale {
  transform: scale(1.3);
  color: rgba(250, 61, 127, 0.39);
}
.qr-code {
  display: flex;
  justify-content: center;
  align-items: center;
}
.qr-code > img {
  width: 100%;
}

.scan-to-join {
  margin-right: 30px;
}

.fullscreenFontSize {
  font-size: 1.5em;
}

/* Full screen */
#current-sld:-moz-full-screen {
  width: 100%;
  background-color: white;
}

#current-sld:-webkit-full-screen {
  width: 100%;
  background-color: white;
}

#current-sld:-ms-fullscreen {
  background-color: white;
  width: 100%; /* needed to center contents in IE */
}
#current-sld:fullscreen {
  width: 100%;
  background-color: white;
  font-size: 3.3em;
  padding: 60px 70px;
}

/* Heading Slide */
.heading-render-container {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  width: 100%;
  height: 100%;
}
.heading-render-container > .heading-render {
  font-size: 1.5em;
  word-break: break-all;
}

/* Open Ended */
#open-sld-content {
  height: 90%;
  overflow-y: scroll;
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  align-content: flex-start;
  font-size: 1.2rem;
}

.open-ended-item {
  width: 31%;
  height: 25%;
  border: 2px solid #7b9085;
  border-radius: 4px;
  overflow-y: scroll;
  margin: 0 1% 15px;
  padding: 5px;
  display: flex;

  justify-content: center;
  /* prettier-ignore */
  box-shadow: 0px 2px 1px -1px rgba(0, 0, 0, 0.2),0px 1px 1px 0px rgba(0, 0, 0, 0.14),0px 1px 3px 0px rgba(0, 0, 0, 0.12);
}

#current-sld:fullscreen > #open-sld-content {
  margin-top: 2%;
}
#current-sld:fullscreen > #open-sld-content > .open-ended-item {
  font-size: 2.5rem;
  padding: 20px;
  margin-bottom: 2%;
}

/* Tag Cloud */
#cloud-sld-content {
  position: absolute;
  left: 50%;
  transform: translateX(-50%);
  width: 80%;
  height: 80%;
}

#current-sld:fullscreen > #cloud-sld-content {
  position: absolute;
  left: 50%;
  transform: translateX(-50%);
  height: 100%;
  width: 100%;
}

/* prev slide/next sld button */
.prev,
.next {
  cursor: pointer;
  position: absolute;
  top: 50%;
  width: auto;
  margin-top: -22px;
  padding: 16px;
  color: white;
  font-weight: bold;
  font-size: 18px;
  transition: 0.6s ease;
  border-radius: 0 3px 3px 0;
  user-select: none;
}

.prev {
  left: 0;
}

.next {
  right: 0;
  border-radius: 3px 0 0 3px;
}

.prev:hover,
.next:hover {
  background-color: rgba(169, 167, 167, 0.8);
}

#current-sld:fullscreen > .prev,
#current-sld:fullscreen > .next {
  display: none;
}

/* at preview size*/

@media (max-width: 1025px) {
  .prev,
  .next {
    background-color: rgba(169, 167, 167, 0.8);
  }
}
@media (max-width: 720px) {
  #current-sld-container {
    height: 100vmin;
    width: 100%;
    max-width: 100vmax;
  }
  #current-sld-border {
    height: 100%;
    padding-top: 0;
    border-radius: 0;
  }
  #current-sld {
    border-radius: 0;
  }
}
