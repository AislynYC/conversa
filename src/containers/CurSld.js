import React, {useEffect, useRef, Fragment, useState} from "react";
import {FormattedMessage} from "react-intl";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import Loading from "../components/Loading/Loading";
import colors from "../lib/colors";

// for each slide types
import Chart from "react-google-charts";
import ReactWordcloud from "react-wordcloud";
import QRCode from "qrcode.react";

const CurSld = props => {
  const clickFullscreen = () => {
    if (
      props.isFullscreen === true &&
      props.slds !== undefined &&
      props.curSldIndex !== undefined
    ) {
      props.nextSld();
    }
  };

  let sldContent = null;
  if (props.slds[props.curSldIndex].sldType === "heading-page") {
    sldContent = <Headings {...props} />;
  } else if (props.slds[props.curSldIndex].sldType === "multiple-choice") {
    sldContent = <MultiSel {...props} colors={colors} />;
  } else if (props.slds[props.curSldIndex].sldType === "open-ended") {
    sldContent = <OpenEnded {...props} />;
  } else if (props.slds[props.curSldIndex].sldType === "tag-cloud") {
    sldContent = <TagCloud {...props} />;
  }

  let handRaised = null;
  const sldRespondedAudi = props.respondedAudi[props.slds[props.curSldIndex].id];
  if (props.slds[props.curSldIndex].sldType !== "heading-page") {
    handRaised = (
      <div className="hand-group">
        <FontAwesomeIcon icon={["fas", "hand-paper"]} id="hand-icon" />
        {sldRespondedAudi ? sldRespondedAudi.length : "0"}
      </div>
    );
  } else {
    handRaised = "";
  }

  return (
    <div id="current-sld-container">
      <div id="current-sld-border">
        <div id="current-sld" onClick={clickFullscreen}>
          {sldContent}
          <div className="member-info">
            {handRaised}
            <FontAwesomeIcon icon={["fas", "user"]} id="member-icon" />
            {props.involvedAudi.length}
          </div>
        </div>
      </div>
    </div>
  );
};

export default CurSld;

const Headings = props => {
  let headingContainer = null;
  let scanToJoinClass = "scan-to-join";
  let qrCodeSize = 230;
  let qrCodeImgSize = {height: 45, width: 45};

  /* different QRCode size depends on fullscreen or not */
  if (props.isFullscreen) {
    scanToJoinClass = "fullscreenFontSize scan-to-join";
    qrCodeSize = 500;
    qrCodeImgSize = {height: 100, width: 100};
  }

  // for switching two types (QRCode or sub-heading) of heading slide
  if (props.slds[props.curSldIndex].hasQRCode) {
    headingContainer = (
      <div className="qr-code">
        <Fragment>
          <div className={scanToJoinClass}>
            <FormattedMessage id="edit.scan-to-join" />
            <FontAwesomeIcon icon={["far", "hand-point-right"]} size="lg" />
          </div>
          <QRCode
            value={`https://conversa-a419b.firebaseapp.com/audi/${props.projId}`}
            size={qrCodeSize}
            imageSettings={{
              src:
                "https://lh3.googleusercontent.com/ZgZLV4rnkakeFPtT14X_lz3BdDpv8kEQ6bzWvXgHw-Wj_WYqiPNSqkq2oUBBRMQiQePrmDmO6WBrRq7_bqFsQBvnTd1_vh4BVg7opHZRsvYUTRrgoL59qyzcflYMnmHy0NzjtngT4A=w400",
              x: null,
              y: null,
              height: qrCodeImgSize.height,
              width: qrCodeImgSize.width,
              excavate: true
            }}
          />
        </Fragment>
      </div>
    );
  } else {
    headingContainer = (
      <div className="sub-heading-render">{props.slds[props.curSldIndex].subHeading}</div>
    );
  }

  return (
    <div className="heading-render-container">
      <div className="heading-render">{props.slds[props.curSldIndex].heading}</div>
      <Fragment>{headingContainer}</Fragment>
    </div>
  );
};

const MultiSel = props => {
  let optsArray = props.slds[props.curSldIndex].opts;
  let resultArray = props.slds[props.curSldIndex].result;
  let resultType = props.slds[props.curSldIndex].resType;
  let optResult = null;
  let pieColors = null;
  if (optsArray !== "") {
    pieColors = props.slds[props.curSldIndex].opts.map((opt, index) => {
      return {color: props.colors[index]};
    });
  }

  if (optsArray !== "") {
    optResult = props.slds[props.curSldIndex].opts.map((opt, index) => {
      let result = resultArray[index] !== "" ? resultArray[index] : 0;
      return [`${opt}`, result, props.colors[index], result];
    });
  } else {
    optResult = null;
  }

  let data = [
    [
      "Options",
      "Result",
      {role: "style"},
      {
        sourceColumn: 0,
        role: "annotation",
        type: "string",
        calc: "stringify"
      }
    ]
  ].concat(optResult);

  let barOptions = {
    legend: {position: "none"},
    height: "100%",
    chartArea: {width: "80%", height: "75%"},
    bar: {groupWidth: "68%"},
    animation: {
      duration: 1000,
      easing: "out",
      startup: true
    },
    vAxis: {
      gridlines: {count: 0},
      minorGridlines: {count: 0},
      ticks: []
    },
    hAxis: {
      textStyle: {
        fontSize: 20
      }
    },
    annotations: {
      textStyle: {
        fontSize: 20,
        bold: true
      }
    }
  };

  let pieOptions = {
    slices: pieColors,
    animation: {
      duration: 1000,
      easing: "out",
      startup: true
    },
    is3D: false,
    pieHole: 0,
    legend: {
      position: "labeled",
      textStyle: {
        fontSize: 16
      }
    },
    pieSliceText: "value",
    pieSliceTextStyle: {
      color: "#333",
      fontSize: 16
    }
  };

  if (props.isFullscreen === true) {
    barOptions = {
      legend: {position: "none"},
      chartArea: {width: "80%", height: "70%"},
      bar: {groupWidth: "68%"},
      animation: {
        duration: 1000,
        easing: "out",
        startup: true
      },
      vAxis: {
        gridlines: {count: 0},
        minorGridlines: {count: 0},
        ticks: []
      },
      hAxis: {
        textStyle: {
          fontSize: 36
        }
      },
      annotations: {
        textStyle: {
          fontSize: 36,
          bold: true
        }
      }
    };
    pieOptions = {
      slices: pieColors,
      animation: {
        duration: 1000,
        easing: "out",
        startup: true
      },
      is3D: false,
      pieHole: 0,
      legend: {
        position: "labeled",
        textStyle: {
          fontSize: 36
        }
      },
      pieSliceText: "value",
      pieSliceTextStyle: {
        color: "#333",
        fontSize: 36
      }
    };
  }

  // To make sure the chart will be drawn only when there is an option exists
  let chart = null;
  if (optsArray !== "") {
    if (resultType === "bar-chart") {
      chart = (
        <Chart
          chartType="ColumnChart"
          width="100%"
          height="100%"
          data={data}
          loader={<Loading {...props} />}
          options={barOptions}
        />
      );
    } else if (resultType === "pie-chart") {
      chart = (
        <Chart
          width="100%"
          height="100%"
          chartType="PieChart"
          loader={<Loading {...props} />}
          data={data}
          options={pieOptions}
        />
      );
    }
  }

  return (
    <Fragment>
      <div className="qus-div">{props.slds[props.curSldIndex].qContent}</div>
      {chart}
    </Fragment>
  );
};

const OpenEnded = props => {
  let openEndedResContent = null;

  if (
    props.slds[props.curSldIndex].openEndedRes !== [] &&
    props.slds[props.curSldIndex].openEndedRes !== undefined
  ) {
    openEndedResContent = props.slds[props.curSldIndex].openEndedRes.map(
      (item, index) => {
        return (
          <div className="open-ended-item" key={index} title={item}>
            <p>{item}</p>
          </div>
        );
      }
    );
  }
  return (
    <Fragment>
      <div className="qus-div">{props.slds[props.curSldIndex].qContent}</div>
      <div id="open-sld-content">{openEndedResContent}</div>
    </Fragment>
  );
};

const TagCloud = props => {
  let tagCloudContent = null;
  let tagResArray = Object.entries(props.slds[props.curSldIndex].tagRes);
  let cloudData = tagResArray.map(item => {
    return {text: item[0].toString(), value: item[1]};
  });
  let cloudOptions = {
    colors: colors,
    enableTooltip: true,
    deterministic: true,
    fontFamily: "Noto Sans TC",
    fontSizes: [20, 60],
    fontStyle: "normal",
    fontWeight: "normal",
    padding: 1,
    rotations: 3,
    rotationAngles: [0, 0],
    scale: "log",
    spiral: "archimedean",
    transitionDuration: 1000
  };
  if (props.isFullscreen === true) {
    cloudOptions = {
      colors: colors,
      enableTooltip: true,
      deterministic: true,
      fontFamily: "Noto Sans TC",
      fontSizes: [30, 100],
      fontStyle: "normal",
      fontWeight: "normal",
      padding: 1,
      rotations: 3,
      rotationAngles: [0, 0],
      scale: "log",
      spiral: "archimedean",
      transitionDuration: 1000
    };
  }
  tagCloudContent = <ReactWordcloud options={cloudOptions} words={cloudData} />;
  return (
    <Fragment>
      <div className="qus-div">{props.slds[props.curSldIndex].qContent}</div>
      <div id="cloud-sld-content">{tagCloudContent}</div>
    </Fragment>
  );
};
