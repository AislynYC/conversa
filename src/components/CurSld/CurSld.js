import React, {Fragment, useState, useEffect} from "react";
import {FormattedMessage} from "react-intl";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import Loading from "../Loading/Loading";
import colors from "../../lib/colors";
import "./curSld.css";

// for each slide types
import Chart from "react-google-charts";
import ReactWordcloud from "react-wordcloud";
import QRCode from "qrcode.react";

const CurSld = props => {
  let curSldType = props.slds[props.curSldIndex].sldType;
  let sldRespondedAudi = props.respondedAudi[props.slds[props.curSldIndex].id];
  let [isFullscreen, setIsFullscreen] = useState(false);
  useEffect(() => {
    const monitorFullscreen = () => {
      if (document.fullscreenElement) {
        setIsFullscreen(true);
      } else {
        setIsFullscreen(false);
      }
    };

    document.addEventListener("fullscreenchange", monitorFullscreen);

    return () => {
      document.removeEventListener("fullscreenchange", monitorFullscreen);
    };
  });

  const clickFullscreen = () => {
    if (
      isFullscreen === true &&
      props.slds !== undefined &&
      props.curSldIndex !== undefined
    ) {
      props.switchNextSld();
    }
  };

  let sldContent = null;
  if (curSldType === "heading-page") {
    sldContent = <Headings {...props} isFullscreen={isFullscreen} />;
  } else if (curSldType === "multiple-choice") {
    sldContent = <MultiSel {...props} colors={colors} isFullscreen={isFullscreen} />;
  } else if (curSldType === "open-ended") {
    sldContent = <OpenEnded {...props} isFullscreen={isFullscreen} />;
  } else if (curSldType === "tag-cloud") {
    sldContent = <TagCloud {...props} isFullscreen={isFullscreen} />;
  }

  let handRaised = null;

  if (curSldType !== "heading-page") {
    handRaised = (
      <div className="hand-group">
        <FontAwesomeIcon icon={["fas", "hand-paper"]} id="hand-icon" />
        {sldRespondedAudi ? sldRespondedAudi.length : "0"}
      </div>
    );
  } else {
    handRaised = "";
  }

  return (
    <div id="current-sld-container">
      <div id="current-sld-border">
        <div id="current-sld" onClick={clickFullscreen}>
          {sldContent}
          <div className="member-info">
            {handRaised}
            <FontAwesomeIcon icon={["fas", "user"]} id="member-icon" />
            {props.involvedAudi.length}
          </div>
          <a className="prev" onClick={props.switchPrevSld}>
            &#10094;
          </a>
          <a className="next" onClick={props.switchNextSld}>
            &#10095;
          </a>
        </div>
      </div>
    </div>
  );
};

export default CurSld;

const Headings = props => {
  let curSld = props.slds[props.curSldIndex];
  let headingContainer = null;
  let scanToJoinClass = "scan-to-join";
  let qrCodeSize = 230;
  let qrCodeImgSize = {height: 45, width: 45};

  /* different QRCode size depends on fullscreen or not */
  if (props.isFullscreen) {
    scanToJoinClass = "fullscreenFontSize scan-to-join";
    qrCodeSize = 500;
    qrCodeImgSize = {height: 100, width: 100};
  }

  // for switching two types (QRCode or sub-heading) of heading slide
  if (curSld.hasQRCode) {
    headingContainer = (
      <div className="qr-code">
        <Fragment>
          <div className={scanToJoinClass}>
            <FormattedMessage id="edit.scan-to-join" />
            <FontAwesomeIcon icon={["far", "hand-point-right"]} size="lg" />
          </div>
          <QRCode
            value={`https://conversa-a419b.firebaseapp.com/audi/${props.projId}`}
            size={qrCodeSize}
            imageSettings={{
              src:
                "https://firebasestorage.googleapis.com/v0/b/conversa-a419b.appspot.com/o/logoC_nb.png?alt=media&token=444f79ff-ef6f-4bfd-a379-316da82b2be2",
              x: null,
              y: null,
              height: qrCodeImgSize.height,
              width: qrCodeImgSize.width,
              excavate: true
            }}
          />
        </Fragment>
      </div>
    );
  } else {
    headingContainer = <div className="sub-heading-render">{curSld.subHeading}</div>;
  }
  const [reactionClass, setReactionClass] = useState("reaction-icon");
  useEffect(() => {
    setReactionClass("reaction-icon scale");
    const timer = setTimeout(() => {
      setReactionClass("reaction-icon");
    }, 300);
    return () => {
      clearTimeout(timer);
    };
  }, [props.reaction.laugh]);
  return (
    <div className="heading-render-container">
      <div className="heading-render">{curSld.heading}</div>
      <Fragment>{headingContainer}</Fragment>
      <div className="reaction-icons">
        <FontAwesomeIcon icon={["fas", "heart"]} className={reactionClass} />
        <span className="reaction-count">{props.reaction.laugh}</span>
      </div>
    </div>
  );
};

const MultiSel = props => {
  let curSld = props.slds[props.curSldIndex];
  let optsArray = curSld.opts;
  let resultArray = curSld.result;
  let resultType = curSld.resType;
  let optResult = null;
  let pieColors = null;

  // build color array structure for pie chart
  if (optsArray !== "") {
    pieColors = optsArray.map((opt, index) => {
      return {color: props.colors[index]};
    });
  }

  if (optsArray !== "") {
    optResult = optsArray.map((opt, index) => {
      let result = resultArray[index] !== "" ? resultArray[index] : 0;
      return [`${opt}`, result, props.colors[index], result];
    });
  } else {
    optResult = null;
  }

  let data = [
    [
      "Options",
      "Result",
      {role: "style"},
      {
        sourceColumn: 0,
        role: "annotation",
        type: "string",
        calc: "stringify"
      }
    ]
  ].concat(optResult);

  let barVar = {
    height: "75%",
    hAxisFontSize: 18,
    annotationsFontSize: 18
  };

  let pieVar = {
    legendFontSize: 16,
    pieSliceFontSize: 16
  };

  if (props.isFullscreen === true) {
    barVar = {
      height: "70%",
      hAxisFontSize: 36,
      annotationsFontSize: 36
    };
    pieVar = {
      legendFontSize: 28,
      pieSliceFontSize: 36
    };
  }

  let barOptions = {
    legend: {position: "none"},
    height: "100%",
    chartArea: {width: "80%", height: barVar.height},
    bar: {groupWidth: "68%"},
    animation: {
      duration: 1000,
      easing: "out",
      startup: true
    },
    vAxis: {
      gridlines: {count: 0},
      minorGridlines: {count: 0},
      ticks: []
    },
    hAxis: {
      textStyle: {
        fontSize: barVar.hAxisFontSize
      }
    },
    annotations: {
      textStyle: {
        fontSize: barVar.annotationsFontSize,
        bold: true
      }
    }
  };

  let pieOptions = {
    slices: pieColors,
    animation: {
      duration: 1000,
      easing: "out",
      startup: true
    },
    is3D: false,
    pieHole: 0,
    legend: {
      position: "labeled",
      textStyle: {
        fontSize: pieVar.legendFontSize
      }
    },
    pieSliceText: "value",
    pieSliceTextStyle: {
      color: "#333",
      fontSize: pieVar.pieSliceFontSize
    }
  };

  // To make sure the chart will be drawn only when there is an option exists
  let chart = null;
  if (optsArray !== "") {
    if (resultType === "bar-chart") {
      chart = (
        <Chart
          chartType="ColumnChart"
          width="100%"
          height="100%"
          data={data}
          options={barOptions}
          loader={<Loading {...props} />}
        />
      );
    } else if (resultType === "pie-chart") {
      chart = (
        <Chart
          chartType="PieChart"
          width="100%"
          height="100%"
          data={data}
          options={pieOptions}
          loader={<Loading {...props} />}
        />
      );
    }
  }

  return (
    <Fragment>
      <div className="qus-div">{curSld.qContent}</div>
      {chart}
    </Fragment>
  );
};

const OpenEnded = props => {
  let curSld = props.slds[props.curSldIndex];
  let openEndedResContent = null;

  if (curSld.openEndedRes !== [] && curSld.openEndedRes !== undefined) {
    openEndedResContent = curSld.openEndedRes.map((item, index) => {
      return (
        <div className="open-ended-item" key={index} title={item}>
          <p>{item}</p>
        </div>
      );
    });
  }
  return (
    <Fragment>
      <div className="qus-div">{curSld.qContent}</div>
      <div id="open-sld-content">{openEndedResContent}</div>
    </Fragment>
  );
};

const TagCloud = props => {
  let curSld = props.slds[props.curSldIndex];
  let tagResArray = Object.entries(curSld.tagRes);
  let cloudData = tagResArray.map(item => {
    return {text: item[0].toString(), value: item[1]};
  });
  let cloudVar = {
    fontSizes: [20, 60]
  };
  if (props.isFullscreen) {
    cloudVar = {
      fontSizes: [30, 100]
    };
  }
  let cloudOptions = {
    colors: colors,
    enableTooltip: true,
    deterministic: true,
    fontFamily: "Noto Sans TC",
    fontSizes: cloudVar.fontSizes,
    fontStyle: "normal",
    fontWeight: "normal",
    padding: 1,
    rotations: 3,
    rotationAngles: [0, 0],
    scale: "log",
    spiral: "archimedean",
    transitionDuration: 1000
  };

  return (
    <Fragment>
      <div className="qus-div">{curSld.qContent}</div>
      <div id="cloud-sld-content">
        <ReactWordcloud options={cloudOptions} words={cloudData} />
      </div>
    </Fragment>
  );
};

export {Headings, MultiSel, OpenEnded, TagCloud};
