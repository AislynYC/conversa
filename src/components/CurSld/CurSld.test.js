import React from "react";
import Enzyme, {shallow} from "enzyme";
import CurSld from "./CurSld";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({adapter: new Adapter()});

const mockSlds = [
  {
    hasQRCode: true,
    heading: "TestHeading 測試標題",
    id: 1583380834131,
    lastEdited: 1583652738079,
    openEndedRes: [],
    opts: ["", "", ""],
    qContent: "",
    resType: "bar-chart",
    result: ["", "", ""],
    sldType: "heading-page",
    subHeading: "",
    tagNum: 1,
    tagRes: {}
  },
  {
    hasQRCode: false,
    heading: "TestHeading 測試標題",
    id: 1583380855588,
    lastEdited: 1583652738080,
    openEndedRes: [],
    opts: ["testOpt1 測試選項１", "testOpt2 測試選項２", "testOpt3 測試選項 ３"],
    qContent: "test question 測試問題",
    resType: "bar-chart",
    result: ["2", "7", "10"],
    sldType: "multiple-choice",
    subHeading: "",
    tagNum: 1,
    tagRes: {}
  }
];

const mockRespondedAudi = {
  1583380834131: [],
  1583380855588: ["043ab6f1-f8da-4188-890e-afab4487772c"],
  1583380867834: [
    "12444937-ef1c-413b-a1da-65be1947b678",
    "dd03c865-af76-4c72-8650-e27e036779d3"
  ]
};

const mockInvolvedAudi = [
  "043ab6f1-f8da-4188-890e-afab4487772c",
  "12444937-ef1c-413b-a1da-65be1947b678",
  "dd03c865-af76-4c72-8650-e27e036779d3"
];

const headingProps = {
  projId: "OJwi11d8aVOTKn6tP0A3",
  curSldIndex: 0,
  slds: mockSlds,
  respondedAudi: mockRespondedAudi,
  involvedAudi: mockInvolvedAudi,
  isFullscreen: true
};

const multiChoiceProps = {
  projId: "OJwi11d8aVOTKn6tP0A3",
  curSldIndex: 1,
  slds: mockSlds,
  respondedAudi: mockRespondedAudi,
  involvedAudi: mockInvolvedAudi,
  isFullscreen: false
};

// Test for slide type: Heading-Page
describe("<CurSld />", () => {
  const wrapper = shallow(<CurSld {...headingProps} />);
  it("match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("must have member info", () => {
    expect(wrapper.find(".member-info").children()).toHaveLength(2);
    expect(
      wrapper
        .find(".member-info")
        .childAt(1)
        .text()
    ).toEqual("3");
  });
});

// Test for slide type: Multi-Choice
describe("<CurSld />", () => {
  const wrapper = shallow(<CurSld {...multiChoiceProps} />);
  it("match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("must have correct member info", () => {
    expect(wrapper.find(".member-info").children()).toHaveLength(3);
    expect(
      wrapper
        .find(".member-info")
        .childAt(2)
        .text()
    ).toEqual("3");
  });

  it("must have correct hand-raised info", () => {
    expect(
      wrapper
        .find(".hand-group")
        .childAt(1)
        .text()
    ).toEqual("1");
  });
});
