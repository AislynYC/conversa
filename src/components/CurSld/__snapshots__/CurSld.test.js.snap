// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurSld /> match snapshot 1`] = `ShallowWrapper {}`;

exports[`<CurSld /> match snapshot 2`] = `ShallowWrapper {}`;
